R/indexing.R

```diff
@@ -190,34 +220,20 @@
       if (any(res != 0)) {
         res[res != 0] <- edge_attr(x, attr, res[res != 0])
       }
+    } else if (!is.null(weights)) {
+      # TODO how to use weights here?!
```

Comment:
First, I am not familiar with this part of the code, and I haven't really used the `g[ ... ]` syntax to treat the graph as a matrix.

But it seems to me that the problem is what I pointed out when the `multi` argument of `get.edge.ids()` was deprecated because it wouldn't function anymore.

We need a function that _can_ return multiple edges between two vertices correctly, and we don't have that now. We need an interface to `igraph_get_all_eids_between()` first, before you can proceed here.

This is why I keep protesting about doing partial upgrades, removing functionality (due to changes in C) but without exposing an interface to the replacement.

Reply:
It might be necessary (for reasonable performance) to implement part of this in pure C, and perhaps add it to the C core directly. It's late here now and I can't quite see how to best do this. Let me get back to you later.

Reply:
https://github.com/igraph/igraph/issues/2476